lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -418,10 +418,10 @@ def add_3ds(post, options)
             post[:additionalData][:scaExemption] = options[:sca_exemption] if options[:sca_exemption]
           end
         else
-          return unless options[:execute_threed] || options[:threed_dynamic]
+          return unless !options[:execute_threed].nil? || !options[:threed_dynamic].nil?
 
-          post[:browserInfo] = { userAgent: options[:user_agent], acceptHeader: options[:accept_header] }
-          post[:additionalData] = { executeThreeD: 'true' } if options[:execute_threed]
+          post[:browserInfo] = { userAgent: options[:user_agent], acceptHeader: options[:accept_header] } if options[:execute_threed] || options[:threed_dynamic]
+          post[:additionalData] = { executeThreeD: options[:execute_threed] } if options[:execute_threed]
```

Comment:
`if options[:execute_threed]` evaluates to false here, so the field is not added. I tested using the !nil? pattern you did on line 421, which works.

Reply:
Oh right, I keep assuming `if options[:execute_threed]` is going to return truthy if the value is present.